pool/pool_machine.c

```diff
@@ -720,7 +720,23 @@ M0_INTERNAL int m0_poolmach_state_transit(struct m0_poolmach       *pm,
 					event->pe_state);
 		m0_poolmach_event_list_dump_locked(pm);
 	}
-	pm->pm_pver->pv_is_dirty = state->pst_nr_failures > 0;
+
+	/*
+	 * Skipping to mark pool version as dirty in case of dix/md pool.
+	 * Assuming DIX and MD has N=1 always
+	 */
+	if (pm->pm_pver->pv_attr.pa_N != 1) {
+		M0_LOG(M0_DEBUG, "pver is marked dirty: "FID_F,
+		       FID_P(&pm->pm_pver->pv_id));
+		pm->pm_pver->pv_is_dirty = state->pst_nr_failures > 0;
+	} else {
+		pm->pm_pver->pv_is_dirty = state->pst_nr_failures > 
+					   pm->pm_pver->pv_attr.pa_K;
```

Comment:
Please check if the functions like `dix_rop_failures_analyse` become obsolete. As per this patch, a pool version is marked as dirty only when we have  more than K failures; but in this case we do not have to send anything because we don't have enough alive services (see `dix_rop_max_failures`). With this change, we do not need to analyse failures when pool version is dirty, we can fail (EIO) DIX request right away.